### PR TITLE
Chen

### PR DESCRIPTION
**Pull request by chen0899**
add feedback

### CODE REVIEW

File: src/main/java/com/softserve/demo/controller/FeedbackController.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    private final FeedbackService feedbackService;
+
+
+    public FeedbackController(FeedbackService feedbackService) {
+        this.feedbackService = feedbackService;
+    }
+
+    @GetMapping("find-all")
+    @ResponseStatus(HttpStatus.OK)
```
Comment (SilichAlex) [inline src/main/java/com/softserve/demo/controller/FeedbackController.java]: This is default value

File: src/main/java/com/softserve/demo/controller/ProvidersController.java [hunk 2]
```diff
@@ -39,92 +32,74 @@ public ProvidersController(ProvidersService providersService, FilesStorageServic
...
-        locationDTO.setRegion(providerAndLocationDTO.getRegion());
-        providerDTO.setLocation(locationDTO);
-        return providerDTO;
-    }
 
     @PostMapping("save")
-    public ResponseEntity<?> saveServiceProvider(@RequestBody ProviderAndLocationDTO providerAndLocationDTO) {
-        ProviderDTO providerDTO = getProviderDTO(providerAndLocationDTO);
-        return new ResponseEntity<>(providersService.save(providerDTO,providerDTO.getLocation()), HttpStatus.OK);
+    @ResponseStatus(HttpStatus.OK)
```
Comment (SilichAlex) [inline src/main/java/com/softserve/demo/controller/ProvidersController.java]: The same case

File: src/main/java/com/softserve/demo/service/impl/FeedbackServiceImpl.java [hunk 3]
```diff
@@ -1,31 +1,100 @@
...
+    }
+
+
+    @Override
+    public FeedbackDTO save(FeedbackDTO feedbackDTO) {
+        Feedback feedback = feedbackMapper.feedbackDTOToFeedback(feedbackDTO);
+        feedback.setRating(0.);
+        feedback.setAddressedFrom(userRepository.findById(feedback.getAddressedFrom().getId()));
+        feedback.setAddressedTo(userRepository.findById(feedback.getAddressedTo().getId()));
+        LocalDateTime today = LocalDateTime.now();
```
Comment (SilichAlex) [inline src/main/java/com/softserve/demo/service/impl/FeedbackServiceImpl.java]: You don`t need to use today valiable

File: src/main/java/com/softserve/demo/service/impl/FeedbackServiceImpl.java [hunk 4]
```diff
@@ -1,31 +1,100 @@
...
+        feedback.setCreatedDate(today);
+        LocalDateTime endDate = today.plusMinutes(10);
```
Comment (SilichAlex) [inline src/main/java/com/softserve/demo/service/impl/FeedbackServiceImpl.java]: The same

File: src/main/java/com/softserve/demo/service/impl/FeedbackServiceImpl.java [hunk 5]
```diff
@@ -1,31 +1,100 @@
...
+        feedback.setEndDate(endDate);
+        feedback.setUpdateDate(null);
+        feedback.setUserTo(feedback.getAddressedTo().getUsername() + ", " + feedback.getAddressedTo().getEmail());
```
Comment (SilichAlex) [inline src/main/java/com/softserve/demo/service/impl/FeedbackServiceImpl.java]: You don`t need this function

File: src/main/java/com/softserve/demo/service/impl/FeedbackServiceImpl.java [hunk 6]
```diff
@@ -1,31 +1,100 @@
...
+        feedback.setUserFrom(feedback.getAddressedFrom().getUsername() + ", " + feedback.getAddressedFrom().getEmail());
+        feedbackRepository.save(feedback);
+        return feedbackMapper.feedbackToFeedbackDTO(feedback);
+    }
+
+    @Override
+    public FeedbackDTO update(FeedbackDTO feedbackDTO) {
+        Feedback feedback = feedbackMapper.feedbackDTOToFeedback(feedbackDTO);
+        Feedback newFeedback = feedbackRepository.findById(feedback.getId()).get();
```
Comment (SilichAlex) [inline src/main/java/com/softserve/demo/service/impl/FeedbackServiceImpl.java:55]: What is this?

File: src/main/java/com/softserve/demo/controller/FeedbackController.java (hunk 1 above)
Comment (chen0899) [reply src/main/java/com/softserve/demo/controller/FeedbackController.java]: fixed

File: src/main/java/com/softserve/demo/controller/ProvidersController.java (hunk 2 above)
Comment (chen0899) [reply src/main/java/com/softserve/demo/controller/ProvidersController.java]: fixed

**Review by serhiylev — APPROVED** [review]

**Review by AnastasiiaHanuliak — APPROVED** [review]

**Pull request merged**